### PR TITLE
Windows: Add SettingsPlugin to add theme change support in win32

### PR DESCRIPTION
**Pull request by moko256**
This PR adds `SettingsPlugin` to add theme change support in win32.

In the #28131 and flutter/flutter#88520, the top-level window message is used to notify brightness change, but I found I cannot use with the same pattern to notify text scale factor change.
In this PR, I separate sending flutter/settings as `SettingsPlugin`, remove the functions to notify brightness change from out of engine, and add watching brightness change in `SettingsPlugin` with registry when win32.

This PR will remove some public functions added in #28131, but that is only used in flutter/flutter#88520 that is not merged, so I think it does not cause large effect.

This PR will close flutter/flutter#88520.
This PR will close flutter/flutter#98355.

No changes in `flutter/tests`.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by moko256** [comment]
I resolved the conflicts and this is ready for review.

File: shell/platform/windows/event_watcher_win32.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_EVENT_WATCHER_WIN32_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_EVENT_WATCHER_WIN32_H_
+
+#include <Windows.h>
+
+#include <functional>
+
+namespace flutter {
+
+// A win32 `HANDLE` wrapper to use it as one-time callback.
```
Comment (cbracken) [inline shell/platform/windows/event_watcher_win32.h:14]: nit: 
```
// A win32 `HANDLE` wrapper for use as a one-time callback.
```

File: shell/platform/windows/event_watcher_win32.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+class EventWatcherWin32 {
+ public:
+  explicit EventWatcherWin32(std::function<void()> callback);
+  ~EventWatcherWin32();
+
+  // Returns `HANDLE` that can be used to register.
```
Comment (cbracken) [inline shell/platform/windows/event_watcher_win32.h:20]: nit:
```
// Returns `HANDLE`, which can be used to register an event listener.
```

File: shell/platform/windows/settings_plugin.h [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+#include <memory>
+
+#include "flutter/shell/platform/common/client_wrapper/include/flutter/basic_message_channel.h"
+#include "flutter/shell/platform/common/client_wrapper/include/flutter/binary_messenger.h"
+#include "flutter/shell/platform/windows/task_runner.h"
+#include "rapidjson/document.h"
+
+namespace flutter {
+
+// Abstract settings plugin.
```
Comment (cbracken) [inline shell/platform/windows/settings_plugin.h:17]: This is fine, but maybe add a tiny bit of detail here:
```
// Abstract settings plugin.
//
// Used to look up and notify Flutter of user-configured system settings.
// These are typically set in the control panel.
```

**Review by cbracken — APPROVED** [review]
Overall, this looks great. Thank you for the patch.

LGTM stamp from a Japanese personal seal

File: shell/platform/windows/settings_plugin.h [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  virtual ~SettingsPlugin();
+
+  static std::unique_ptr<SettingsPlugin> Create(BinaryMessenger* messenger,
+                                                TaskRunner* task_runner);
+
+  // Sends settings (e.g., platform brightness) to the engine.
+  void SendSettings();
+
+  // Start watching settings change and notify the engine ot the update.
```
Comment (loic-sharma) [inline shell/platform/windows/settings_plugin.h:30]:
```suggestion
  // Start watching settings changes and notify the engine of the update.
```

Comment (moko256) [reply shell/platform/windows/settings_plugin.h]: Done. Thanks!

File: shell/platform/windows/settings_plugin.h (hunk 3 above)
Comment (moko256) [reply shell/platform/windows/settings_plugin.h:17]: Done!

File: shell/platform/windows/event_watcher_win32.h (hunk 2 above)
Comment (moko256) [reply shell/platform/windows/event_watcher_win32.h]: Done!

File: shell/platform/windows/event_watcher_win32.h (hunk 1 above)
Comment (moko256) [reply shell/platform/windows/event_watcher_win32.h]: Done. Thanks!

**Review by dkwingsmt — APPROVED** [review]
LGTM, very clean code!

**Pull request merged**